packages/v2/gui/src/prosess/ung-beregning/ArbeidOgInntekt.tsx

```diff
@@ -33,35 +33,36 @@ const formaterStatus = (status?: KontrollerInntektPeriodeDtoStatus) => {
 };
 
 const buildInitialValues = (
-  inntektKontrollperioder: KontrollerInntektDto['kontrollperioder'],
+  inntektKontrollperioder: Array<KontrollerInntektPeriodeDto>,
   aksjonspunkt: AksjonspunktDto | undefined,
-) => {
-  const vurdertPeriode = inntektKontrollperioder?.find(
-    periode => periode.erTilVurdering && periode.status === KontrollerInntektPeriodeDtoStatus.AVVIK && periode.valg,
-  );
-  if (vurdertPeriode) {
-    return {
-      fastsattInntekt: vurdertPeriode.fastsattInntekt ? `${vurdertPeriode.fastsattInntekt}` : '',
-      valg: (vurdertPeriode.valg as KontrollerInntektPeriodeDtoValg) ?? '',
-      begrunnelse: aksjonspunkt?.begrunnelse ?? '',
-    };
-  }
+): Formvalues => {
   return {
-    fastsattInntekt: '',
-    valg: '' as const,
-    begrunnelse: '',
+    perioder:
+      inntektKontrollperioder
+        ?.filter(periode => periode.erTilVurdering)
+        .map(periode => {
+          return {
+            fastsattInntekt: periode.fastsattInntekt ? `${periode.fastsattInntekt}` : '',
+            valg: (periode.valg as KontrollerInntektPeriodeDtoValg) ?? '',
```

Comment:
Trur ikkje det trengs casting her
```suggestion
            valg: periode.valg ?? '',
```